### ISSUE TITLE
Regent: Attaching causes codegen assertion

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#670]
The following code snippet:

```import "regent"
struct vec {
  v_1 : double
}
struct Node_Vals {
  prim_vars_cc : vec
}
task loadTile(s : region(ispace(int3d), Node_Vals))
where reads writes(s.prim_vars_cc) do
  attach(hdf5, s.prim_vars_cc, 'foo.hdf', regentlib.file_read_only)
  detach(hdf5, s.prim_vars_cc)
end
task main()
end
regentlib.start(main)
```

causes the following error:

```
/home/manolis/sw/legion/language/src/regent/codegen.t:335: assertion failed!
stack traceback:
	[C]: in function 'assert'
	/home/manolis/sw/legion/language/src/regent/codegen.t:335: in function 'userfn'
	/home/manolis/sw/legion/language/src/regent/codegen.t:6961: Errors reported during evaluating Lua code from Terra
               [cx:region(region_type):field_id(field_path)],
                 ^
	src/terralib.lua:1795: in function 'evalluaexpression'
	src/terralib.lua:2819: in function 'docheck'
	src/terralib.lua:3030: in function 'checkexp'
	src/terralib.lua:2557: in function 'checkexpressions'
	src/terralib.lua:2724: in function 'checksingle'
	src/terralib.lua:3232: in function 'checkstmts'
	src/terralib.lua:2992: in function 'docheck'
	...
	src/terralib.lua:1154: in function 'definequote'
	/home/manolis/sw/legion/language/src/regent/codegen.t:6954: in function 'expr'
	/home/manolis/sw/legion/language/src/regent/codegen.t:9732: in function 'fn'
	src/asdl.lua:18: in function 'block'
	/home/manolis/sw/legion/language/src/regent/codegen.t:11001: in function 'compile_thunk'
	/home/manolis/sw/legion/language/src/regent/std_base.t:1472: in function 'complete'
	/home/manolis/sw/legion/language/src/regent/std.t:4276: in function </home/manolis/sw/legion/language/src/regent/std.t:4274>
	/home/manolis/sw/legion/language/src/regent/std.t:4278: in function 'compile_tasks_in_parallel'
	/home/manolis/sw/legion/language/src/regent/std.t:4393: in function 'start'
	simple_hdf_example.rg:15: in main chunk
```

**Comment by magnatelee** [comment]
Should be fixed by ff7d4850.

**Comment by manopapad** [comment]
Fixed
